examples/Example8_SendReceiveBinary/Example8_SendReceiveBinary.ino

```diff
@@ -0,0 +1,139 @@
+        ;
+    notecard.setDebugOutputStream(stlinkSerial);
+#endif
+
+    // Initialize the physical I/O channel to the Notecard
+#ifdef txRxPinsSerial
+    notecard.begin(txRxPinsSerial, 9600);
+#else
+    notecard.begin();
+#endif
+
+    // "notecard.newRequest()" uses the bundled "J" json package to allocate a
+    //"req", which is a JSON object for the request to which we will then add
+    // Request arguments.  The function allocates a "req" request structure
+    // using malloc() and initializes its "req" field with the type of request.
+    J *req = notecard.newRequest("hub.set");
+
+    // This command (required) causes the data to be delivered to the Project on
+    // notehub.io that has claimed this Product ID (see above).
+    JAddStringToObject(req, "product", PRODUCT_UID);
+
+    // This command determines how often the Notecard connects to the service.
+    // "continous" mode is required for binary file transfers
+    JAddStringToObject(req, "mode", "continuous");
+    notecard.sendRequestWithRetry(req, 5); // 5 seconds
+
+    // Reset the Notecard's binary storage area, so that we can start fresh
+    NoteBinaryReset();
+
+    // #####################
+    // OUTBOUND FILE EXAMPLE
+    // #####################
+
+    // Check the binary buffer to make sure the fragment size will fit
+    setup_binary_buffer(CHUNK_SIZE, sizeof(temp_buffer));
+    notecard.logDebugf("Sending image of size %u, in chunks of size: %u\n", img_len, CHUNK_SIZE);
+
+    // send the binary file to Notecard in fragments of size CHUNK_SIZE
+    int i = 0;
+    size_t bytes_left = img_len;
+    while (bytes_left)
+    {
+        notecard.logDebugf("\nSending chunk %d, offset: %d...\n", i, i * CHUNK_SIZE);
+
+        size_t bytes_to_send = bytes_left >= CHUNK_SIZE ? CHUNK_SIZE : bytes_left;
+        memcpy(temp_buffer, img_map + i * CHUNK_SIZE, bytes_to_send);
+
+        if (!send_binary_data_to_notecard(temp_buffer, bytes_to_send))
+        {
+            bytes_left -= bytes_to_send;
+            i++;
+        }
+    }
+
+    notecard.logDebug("File transfered to Notecard, now web.post to Notehub\n");
+    send_binary_data_to_notehub();
+
+    notecard.logDebug("Done with outgoing file!\n");
+
+    // ####################
+    // INBOUND FILE EXAMPLE
+    // ####################
+
+    if (J *req = NoteNewRequest("web.get"))
+    {
+        JAddStringToObject(req, "route", "GetImage");      // update with your route name
+        JAddStringToObject(req, "content", "images/jpeg"); // update with your content type
+        JAddBoolToObject(req, "binary", true);
+
+        if (!NoteRequest(req))
+        {
+            notecard.logDebug("Error receiving image\n");
+        }
+
+        // get length of downloaded binary data
+        size_t buffer_len = 0;
+        NoteBinaryRequiredRxBuffer(&buffer_len);
+
+        // Call NoteBinaryReceive to verify and decode the binary data
+        uint8_t *my_binary_data = (uint8_t *)malloc(buffer_len);
```

Comment:
`my_binary_data` is never freed. Doesn't really matter, but I'd free it for consistency and in case anyone copy-pastes this code verbatim for their own use.

Reply:
Got it